sdk/storage/faq.md

```diff
@@ -0,0 +1,374 @@
+SDK will still set headers and query parameters that are added between the customized API version (which is usually older) and latest API version,
+and still expect some headers or fields in HTTP response.
+Furthermore, this scenario is not covered by testing, although most of the APIs should be able to work. You should do thorough testing before your code goes to production.
+
+## How can I set a custom UserAgent string?
+
+We recommend you set an application ID with code below, so that it can be identified from which application, SDK, platform the request was sent. The information could be useful for troubleshooting and telemetry purposes.
+
+```C++
+BlobClientOptions clientOptions;
+clientOptions.Telemetry.ApplicationId = "SomeApplication v1.2.3";
+
+// The User-Agent string will be something like:
+// SomeApplication v1.2.3 azsdk-cpp-storage-blobs/12.9.0-beta.3 (Windows 10 Enterprise 6.3 19045 19041.1.amd64fre.vb_release.191206-1406)
+```
+
+## What encoding is used for input and output?
+
+All URLs should be URL-encoded, other resource names should be in UTF-8 encoding.
+This applies to both input variables and output.
+If your code runs in an environment where the default locale and encoding is not UTF-8, you should encode before passing variables into SDK and decode after reading a variable from SDK.
+
+```C++
+// containerUrl should be URL-encoded
+auto containerClient = BlobContainerClient(containerUrl);
+// The return value is URL-encoded
+auto url = containerClient.Geturl();
+// blobName should be UTF-8 encoded
+auto blobClient = containerClient.GetBlobClinet(blobName);
+
+for (auto page = blobContainerClient.ListBlobs(); page.HasPage(); page.MoveToNextPage()) {
+  for (auto& blob : page.Blobs) {
+    // blob.Name is UTF-8 encoded
+    std::cout << blob.Name << std::endl;
+  }
+}
+```
+
+## How can I check if a blob exists? Is there a function `Exists()` like in legacy SDK?
+
+There's no equivalent in the new C++ SDK. This function is usually just a convenient method on top of getting blob properties. You could easily write one like below.
+
+```C++
+bool BlobExists(const Azure::Storage::Blobs::BlobClient& client) {
+  try {
+    client.GetProperties();
+    return true;
+  }
+  catch (Azure::Storage::StorageException& e) {
+    if (e.ReasonPhrase == "The specified blob does not exist.") {
+      return false;
+    }
+    throw;
+  }
+}
+```
+
+A common use scenario is checking if a blob exists before writing to it.
+It is fine if you write from a single thread.
+However, it is problematic if the destination might be modified by multiple threads or processes. A subsequent write operation may get overwritten by the previous one.
+In this case, it's more recommended to:
+
+1. Use `CreateIfNotExists()`, `DeleteIfExists()` functions whenever possible. These functions internally use access conditions and can help us catch unexpected exceptions caused by resending PUT/DELETE requests on network errors.
```

Comment:
> Help us catch...

Who is us in this case? The SDK authors, the service authors, or the reader/customer? If it's the reader, change to `you`.